src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -0,0 +1,127 @@
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String propertyKey = entry.getValue();
+
+      // if updating for registering an organization then do not want to consider fields that require maintainer or moderator permissions
+      if(forRegistration && (requiresModerator.contains(propertyKey) || requiresMaintainer.contains(propertyKey))) {
+          continue;
+      }
+      // will only get approval param is a maintainer sent the request
+      if(requiresMaintainer.contains(propertyKey) && !isMaintainer) {
+        continue;
+      }
+
+      // will only get moderatorList param if request is sent by a moderator or maintainer
+      if(requiresModerator.contains(propertyKey) && !(isModerator || isMaintainer)) {
+        continue;
+      }
+
+      String formKey = entry.getKey();
+      String formValue = "";
+      // setting organization description is optional
+      if(!propertyKey.equals("orgDescription")) {
+        try {
+          formValue = getParameterOrThrow(request, formKey);
+        } catch(IllegalArgumentException err) {
+          throw new IllegalArgumentException();
+        }
+      } else if (request.getParameter(formKey) != null) {
+          formValue = request.getParameter(formKey);
+      } else {
+          formValue = "";
+      }
+
+      setOrganizationProperty(propertyKey, formValue);
+    }
+  }
+
+  private String getParameterOrThrow(HttpServletRequest request, String formKey) {
+    String result = request.getParameter(formKey);
+    if(result.isEmpty() || result == null) {
+      throw new IllegalArgumentException("Form value cannot be null");
+    }
+    return result;
+  }
+
+  private void setOrganizationProperty(String propertyKey, String formValue) {
+    if(propertyKey.equals("moderatorList")) {
+      ArrayList<String> newModeratorList = translateEmailsToIds(Arrays.asList(formValue.split("\\s*,\\s*")));
+      this.entity.setProperty("moderatorList", newModeratorList);
+    } else if(propertyKey.equals("isApproved")) {
+      if(formValue.equals("approved")) {
+        this.entity.setProperty("isApproved", true);
+      } else {
+          this.entity.setProperty("isApproved", false);
+      }
+    } else {
+        this.entity.setProperty(propertyKey, formValue);
+    }
+  }
+
+  private ArrayList<String> translateEmailsToIds(ArrayList<String> emails) {
+    ArrayList<String> userIds = new ArrayList<String>();
+    for(String email : emails) {
+      GivrUser newUser = GivrUser.getUserByEmail(email);
```

Comment:
can the user returned ever be null? or can the email address fail to generate a user id (from a typo, for instance!).

Reply:
reading the documentation it seems that it will always make a new user even if a typo exists so that unique id will be added to the list

Reply:
Looks like it can return null

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/users/User.html#getuserid

Reading that, I'm not sure if we can get the user from an email address. Can you file an issue to investigate?